src/Linters/DontHaveTwoEmptyLinesInARowLinter.hack

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+final class DontHaveTwoEmptyLinesInARowLinter extends AutoFixingASTLinter {
+  const type TContext = Token;
+  const type TNode = EndOfLine;
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    this::TContext $token,
+    this::TNode $_i_make_hhast_not_hand_me_nodes_that_dont_have_eol_in_trivia,
+  ): ?ASTLintError {
+    $eol_count = $this->getAST()
+      ->getPreviousToken($token)
+      ?->getTrailing()
+      ?->getLast()
+      |> $$ is EndOfLine ? 1 : 0;
+
+    foreach ($token->getLeading()->toVec() as $trivia) {
+      if ($trivia is EndOfLine) {
+        $eol_count++;
+        if ($eol_count === 3) {
+          return new ASTLintError(
+            $this,
+            "Don't have two empty lines in a row",
+            $token,
+            () ==> static::removeLeading($token, $trivia),
+          );
+        }
+      } else {
+        $eol_count = 0;
+      }
+    }
+
+    // We don't check the boundary with the next token.
+    // This will be checked by the loop above when linting
+    // that token instead.
```

Comment:
What if trailing ends with 2 or more empty lines? I don't know if this can happen.